tests/test-theme-templates.php

```diff
@@ -0,0 +1,114 @@
+	}
+
+	public function test_headings_are_localized() {
+		$template          = new stdClass();
+		$template->content = '
+			<!-- wp:heading -->
+			<h2 class="wp-block-heading">This is a heading to localize.</h2>
+			<!-- /wp:heading -->
+		';
+		$new_template      = Theme_Templates::escape_text_in_template( $template );
+		$this->assertStringContainsString( 'This is a heading to localize.', $new_template->content );
+		$this->assertStringNotContainsString( '<h2 class="wp-block-heading">This is a heading to localize.</h2>', $new_template->content );
+	}
+
+	public function test_eliminate_theme_ref_from_template_part() {
+		$template          = new stdClass();
+		$template->content = '<!-- wp:template-part {"slug":"header","theme":"testtheme"} /-->';
+		$new_template      = Theme_Templates::eliminate_environment_specific_content( $template );
+		$this->assertStringContainsString( '<!-- wp:template-part {"slug":"header"} /-->', $new_template->content );
+	}
+
+	public function test_eliminate_nav_block_ref() {
+		$template          = new stdClass();
+		$template->content = '<!-- wp:navigation {"ref":4} /-->';
+		$new_template      = Theme_Templates::eliminate_environment_specific_content( $template );
+		$this->assertStringContainsString( '<!-- wp:navigation /-->', $new_template->content );
+	}
+
+	public function test_eliminate_nav_block_ref_in_nested_block() {
+		$template          = new stdClass();
+		$template->content = '
+			<!-- wp:group {"layout":{"type":"constrained"}} -->
+			<div class="wp-block-group"><!-- wp:navigation {"ref":4} /--></div>
+			<!-- /wp:group -->
+		';
+		$new_template      = Theme_Templates::eliminate_environment_specific_content( $template );
+		$this->assertStringContainsString( '<!-- wp:navigation /-->', $new_template->content );
+	}
+
+	public function test_eliminate_id_from_image() {
+		$template          = new stdClass();
+		$template->content = '
+			<!-- wp:image {"id":635} -->
+			<figure class="wp-block-image size-large"><img src="http://example.com/file.jpg" alt="" class="wp-image-635"/></figure>
+			<!-- /wp:image -->
+		';
+		$new_template      = Theme_Templates::eliminate_environment_specific_content( $template );
+		$this->assertStringContainsString( '<!-- wp:image -->', $new_template->content );
+		$this->assertStringNotContainsString( '<!-- wp:image {"id":635} -->', $new_template->content );
+		$this->assertStringNotContainsString( 'wp-image-635', $new_template->content );
+	}
+
+	// TODO: I'm not sure of the proper way to format this property for testing or now to cause it to be
+	// added via the Global Styles Panel.
+	// public function test_eliminate_taxQuery_from_query_loop() {
+	// 	$template          = new stdClass();
+	// 	$template->content = '
+	// 	<!-- wp:query {"queryId":43,"query":{"perPage":10,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":true},"metadata":{"categories":["posts"]}} -->
```

Comment:
Here's an example with `taxQuery`:

```
<!-- wp:query {"queryId":20,"query":{"perPage":3,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false,"taxQuery":{"post_tag":[]}}} -->
```

Reply:
This attribute is added if you select a filter based on taxonomies in the block settings for a query block:

<img width="317" alt="image" src="https://github.com/WordPress/create-block-theme/assets/1645628/9400af3e-206e-40c1-8845-1296fd5e3ba1">

Reply:
Thank you @mikachan 
I have added that to the test suite [here](https://github.com/WordPress/create-block-theme/pull/522/commits/1d48804ca0954de85df581ad1bb98f5519b0fd95) (which did what tests are supposed to do and showed me an error. :) )